fem/src/CMakeLists.txt

```diff
@@ -176,7 +176,11 @@ IF(HAVE_ZOLTAN)
   IF(Zoltan_ENABLE_ParMETIS)
     TARGET_LINK_LIBRARIES(elmersolver ${ParMetis_LIBRARIES})
   ENDIF()
-  ADD_DEPENDENCIES(elmersolver ${ZOLTAN_LIBRARY})
+  IF(TARGET Zoltan::zoltan)
+    ADD_DEPENDENCIES(elmersolver Zoltan::zoltan)
```

Comment:
IIUC, `Zoltan::zoltan` is only a target if it is referring to an external dependency. In that case, it is always already built. Adding the dependency here doesn't make sense if I'm not mistaken.

Please, correct me if that is wrong.

Reply:
Maybe, conditionally link to that target in line 175 (a couple of lines above in the same file). That might even be necessary in case the Zoltan headers are installed at a non-default prefix (that doesn't happen to be added already).